catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/model/FilterNodeValueSerializer.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.forms.model;
+
+import com.google.common.primitives.Doubles;
+import com.google.common.primitives.Ints;
+import java.lang.reflect.Field;
+import org.boon.core.reflection.fields.FieldAccess;
+import org.boon.json.serializers.CustomFieldSerializer;
+import org.boon.json.serializers.JsonSerializerInternal;
+import org.boon.primitive.CharBuf;
+
+/**
+ * {@link org.codice.ddf.catalog.ui.forms.model.JsonModel.FilterLeafNode} requires polymorphic
+ * behavior which causes issues with JSON serialization. To avoid values of the following complex
+ * form:
+ *
+ * <p>{@code
+ * "value":{"class":"java.lang.String","value":[r,u,s,s,i,a,n,-,n,e,w,s,-,s,t,o,r,y],"hash":0}}
+ *
+ * <p>all values are treated as Strings and the following {@link CustomFieldSerializer} writes them
+ * out to their proper type.
+ */
+public class FilterNodeValueSerializer implements CustomFieldSerializer {
+  @Override
+  public boolean serializeField(
+      JsonSerializerInternal serializer, Object parent, FieldAccess fieldAccess, CharBuf builder) {
+    Field field = fieldAccess.getField();
+    String name = field.getName();
+
+    boolean shouldManuallySerialize =
+        (name.equals("value") || name.equals("defaultValue"))
```

Comment:
Can `name` be null?  The logic could be switched if so; `"value".equals(name)`

Reply:
Name can't be null according to the reflection APIs (they don't actually specify, so I believe our default assumption is non-null). If a particular `Field` cannot be found, read, accessed, or does not exist, those all result in exception cases, which Boon can handle. 

All that being said, I'll still make this change.